garak/generators/rest.py

```diff
@@ -29,7 +29,8 @@ class RestGenerator(Generator):
     DEFAULT_PARAMS = Generator.DEFAULT_PARAMS | {
         "headers": {},
         "method": "post",
-        "ratelimit_codes": [429],
+        "ratelimit_codes": {429},
+        "skip_codes": set(),
```

Comment:
Changing from `list` to set here may not be the best choice, safe `yaml` and `json` serialization do not really contain `set` logic and configuration is passed via these formats.  In this case the values are `replaced` if found in a configuration files so I would prefer we stick to `list` to ensure we do infer a logical interface that may not be available at runtime.

Reply:
This went through my mind as well. I wanted to signal the set-like nature of how arguments are supplied, but this doesn't jive with what can be supplied via existing config formats. Agree that adding `set` here may encourage code that relies on `set` affordances, will update